os-jdg7-launch/added/partitionPV.sh

```diff
@@ -0,0 +1,209 @@
+      mkdir -p "${INSTANCE_DIR}"
+
+      TERMINATING_FILE="${INSTANCE_DIR}/terminating"
+      TERMINATING=$(cat "${TERMINATING_FILE}" 2>/dev/null)
+      if [ -n "$TERMINATING" ] ; then
+        echo "Attempting to migrate directory: ($INSTANCE_DIR)"
+
+        (
+          flock -n $MIGRATING_FD
+          if [ $? -eq 0 ]; then
+            TERMINATING_TIME=$(stat -c "%Y" "${TERMINATING_FILE}")
+            CURRENT_TIME=$(date +"%s")
+            TIMEOUT=$(expr $MIGRATION_TIMEOUT + $TERMINATING_TIME - $CURRENT_TIME)
+            echo "Waiting for grace period to expire, remaining timeout is ${TIMEOUT} seconds"
+            while : ; do
+              TERMINATING=$(cat "${TERMINATING_FILE}" 2>/dev/null)
+              if [ -z "$TERMINATING" ] ; then
+                echo "Migration cancelled, no longer terminating in directory: ($INSTANCE_DIR)"
+                break
+              else
+                TIMEOUT=$(expr $TIMEOUT - 1)
+                if [ "$TIMEOUT" -gt 0 ] ; then
+                  sleep 1
+                else
+                  break
+                fi
+              fi
+            done
+
+            if [ "$TIMEOUT" -le 0 ] ; then
+              echo "Attempting to obtain lock for directory: ($INSTANCE_DIR)"
+
+              flock -n $LOCK_FD
+              LOCK_STATUS=$?
+
+              if [ $LOCK_STATUS -eq 0 ] ; then
+                echo "Successfully locked directory: ($INSTANCE_DIR)"
+                MIGRATED=true
+
+                flock -u $MIGRATING_FD
+
+                SERVER_DATA_DIR="${INSTANCE_DIR}"
+
+                PID=$!
+
+                trap "echo Received TERM ; echo \"$HOSTNAME\" > \"$TERMINATING_FILE\" ; kill -TERM $PID" TERM
+
+                wait $PID 2>/dev/null
+                STATUS=$?
+                trap - TERM
+                wait $PID 2>/dev/null
+
+                echo "Migration terminated with status $STATUS ($(kill -l $STATUS))"
+
+                if [ "$STATUS" -eq 0 ] ; then
+                  > "$TERMINATING_FILE"
+                elif [ "$STATUS" -eq 255 ] ; then
+                  echo "Server returned 255, changing to 254"
+                  STATUS=254
+                fi
+
+                echo "Releasing lock: ($INSTANCE_DIR)"
+                exit $STATUS
+              fi
+            fi
+          fi
+
+          exit 255
+        ) 200> "${INSTANCE_DIR}/lock" 201> "${INSTANCE_DIR}/migrating" &
+
+        PID=$!
+
+        trap "kill -TERM $PID" TERM
+
+        wait $PID 2>/dev/null
+        STATUS=$?
+        trap - TERM
+        wait $PID 2>/dev/null
+
+        if [ $STATUS -ne 255 -a $STATUS -ne 0 ] ; then
+          break;
+        fi
+      fi
+      COUNT=$(expr $COUNT + 1)
+    else
+      if [ "$MIGRATED" = "false" ] ; then
+        echo "Finished Migration Check cycle, pausing for ${MIGRATION_PAUSE} seconds before resuming"
+        COUNT=1
+        sleep "${MIGRATION_PAUSE}"
+      else
+        MIGRATED=false
+      fi
+    fi
+  done
+}
```

Comment:
This is a complete copy of the a-mq script with two minor changes
- SERVER_DATA_DIR has been changed, it should be ${INSTANCE_DIR}/serverData
- the runMigration part has been removed.

I suspect the original script can be used as-is which was certainly my intention, is there any reason why this was copied?

Reply:
My intention was to remove the runMigration() call which contains the AMQ-specific logic as JDG just cares about the file system partition. So I copied the AMQ drain.sh to JDG migratePV.sh and removed the AMQ specific call. The JDG drain pod calls migratePV.sh not drain.sh (see here https://github.com/jboss-openshift/application-templates/pull/335/files).

Then as you indicated I only changed the SERVER_DATA_DIR value in partitionPV.sh as I couldn't figure out what serverData was and assumed it was something AMQ specific. And removed the call to runMigration() since that's the AMQ specific call that now doesn't exist for JDG. 

Does JDG not need the drain pod and the migratePV() call at all? If that's the case then we should be able to use the existing AMQ logic and eliminate the JDG drain pod completely.